sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/authentication/AzureCredentialFactory.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.management.resources.fluentcore.authentication;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Utility class for authentication.
+ */
+public class AzureCredentialFactory {
+
+    private AzureCredentialFactory() {
+    }
+
+    /**
+     * Creates the credential by environment variables
+     * @return the credential.
+     */
+    public static AzureTokenCredential fromEnvironment() {
+        return new AzureTokenCredentialBuilder()
```

Comment:
My suggestion is not merge credential from environment to AzureTokenCredentialBuilder. This factory class should be the place that make the AzureTokenCredential from environment variables.